src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/AesCcm.Unix.cs

```diff
@@ -11,6 +11,8 @@ public sealed partial class AesCcm
     {
         private byte[] _key;
 
+        public static bool IsSupported { get; } = Interop.OpenSslNoInit.OpenSslIsAvailable;
```

Comment:
What happens on macOS when IsSupported is false and the type is used anyways (since IsSupported is new API and therefore has no callers)?

Do we need to put an "if (!IsSupported) throw new PNSE(yadda yadda)" in ImportKey?

Reply:
Ohhh. ChaChaPoly1305 has that in the constructor but not the GCM/CCM types. I'll add the same to GCM/CCM.